### ISSUE TITLE
🚀 Feature: Modularize the site

### ISSUE THREAD

**Issue by MichaelSolati** [amplication/site#156]
### 🔖 Feature description

Use a SSG to be able to break up the code and make updates easy across this site.

### 🎤 Pitch

- By using a SSG we can break out reusable parts of the site so they can be updated only once.
- With a build process we can apply cool "tricks" to improve the sites performance.
  - Minifying HTML
  - Inlining critical CSS
- Now that generic components and parts of the code are no longer in the same file as the content of the pages.
  - The content of the pages are easier to read and edit.
- We can apply different loading behaviors when in development vs production, like CSS files can be deferred in prod as we inline critical CSS. Same as JS. 
  - Overall this would improve FCP, TTI, TBT, and more!
- Some values can be written into the HTML rather than requiring HTTP requests on client. 
- - Like the version number, it's now outputted into the HTML and not gotten with a get request on client.
- We can use data sources to create HTML
  - The teams page is hardcoded HTML, instead we could just loop over data entries in a JSON to render team members.

### ✌️ Solution

I created a PR for this.

### 🔄️ Alternative

You could manually go in and change every file...

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/amplication/amplication/blob/master/code_of_conduct.md)
